docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,497 @@
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+This guide covers creating a ClusterClass cluster with the CAPD provider.
+</aside>
+
+## Installation
+
+### Common Prerequisites
+This guide is based on having the same pre-requisites as the CAPI [quick start guide].
+That is:
+- Install and setup [kubectl]  and [clusterctl]
+- Install [Kind] and [Docker]
+
+ClusterClass is currently behind a feature gate that needs to be enabled. To do so run:
+
+```bash
+export  EXP_CLUSTER_RESOURCE_SET=true
+export CLUSTER_TOPOLOGY=true
+```
+This ensures that Cluster Topology and Cluster Resource Set are enabled when we initialize our CAPI Providers.
+*Note*: Cluster Resource Set is not required for ClusterClass but is used in this tutorial to set up networking.
+
+A script to set up a Kind cluster pre-configured for CAPD can be found in the hack folder. To set up the cluster run:
+```bash
+./hack/kind-install-for-capd.sh
+clusterctl init --infrastructure docker
+````
+
+Alternatively the quickstart guide for setting up a CAPD enabled cluster can be followed.
+
+#### Creating the ClusterClass and templates
+Now that we have a CAPI Management Cluster with CAPD and Cluster Topology enabled we can create the ClusterClass and the templates needed to stamp clusters.
+
+The below contains six yaml objects which are used to define the cluster:
+
+1) ClusterClass
+1) DockerMachineTemplate
+1) DockerClusterTemplate
+1) DockerMachineTemplate for ControlPlane
+1) KubeadmControlPlaneTemplate
+1) KubeadmConfigTemplate
+
+The ClusterClass - first in the below yaml - contains references to the templates needed to build a full cluster using the shape defined in the ClusterClass.
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: ClusterClass
+metadata:
+  name: cluster-class
+  namespace: default
+spec:
+  controlPlane:
+    metadata:
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+      kind: KubeadmControlPlaneTemplate
+      name: my-cluster-control-plane
+      namespace: default
+    machineInfrastructure:
+      ref:
+        kind: DockerMachineTemplate
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+        name: "my-cluster-control-plane"
+        namespace: default
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+      kind: DockerClusterTemplate
+      name: my-cluster
+      namespace: default
+  workers:
+    machineDeployments:
+    - class: linux-worker
+      template:
+        metadata:
+        bootstrap:
+          ref:
+            kind: KubeadmConfigTemplate
+            apiVersion: bootstrap.cluster.x-k8s.io/v1alpha4
+            name: "my-cluster-md-0"
+        infrastructure:
+          ref:
+            kind: DockerMachineTemplate
+            apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+            name: "md1"
```

Comment:
Should we call this the same as the bootstrap config template? md-0 vs m1 looks not ideal.